### ISSUE TITLE
[RFC] Date picker

### ISSUE THREAD

**Issue by brianespinosa** [microsoft/fluent-ui-react#346]
## Feature Request
This is more of a placeholder for discussion around a date picker component.

### Problem description
Right now there is no styled solution for date picking inside of Stardust. Before work could start on this, because of the complexities, we should have a discussion about what we do and do not want.

- Should our picker allow us to combine time and date?
- Should our picker be hard-tied to one date library or allow you to BYO (moment, etc)
- How do we want to handle intl formatting

Feel free to throw in other wants/needs/dislikes and we can update this list above into an actual spec.

@levithomason also chime in if you feel like a date picker would be beyond the scope of Stardust and would be better served as a plugin/extension implementation instead.

**Comment by levithomason** [comment]
>Should our picker allow us to combine time and date?

I think we should favor vanilla JS date objects.  That means date and time would be supported at the data level, at least.

From a UI perspective, I think we would support the common and necessary use cases.  This would include an input for picking a time all the way through dual calendars for start/end dates.

>Should our picker be hard-tied to one date library or allow you to BYO (moment, etc)

Again, I'd go vanilla JS as the base implementation.  Most users will want to use moment.  Some users may want something else or custom.  For this, I'd propose we allow props or an interface for customizing how parsing is done.

>How do we want to handle intl formatting

Guess what I'm going to say... Vanilla JS!  The browser's date object has everything we need for handling localization and formatting.  I would also start with this and as with the previous reply, allow some props or an interface to be passed to handle parsing in a custom way.

You can see the direction we we're headed in Semantic UI React for the `Datetime` API, including the props and how parsing was handled.  I think this was largely the right direction:

https://github.com/Semantic-Org/Semantic-UI-React/pull/2821/files#diff-b16d96abfb1586d6d540ec890e3504ce

>@levithomason also chime in if you feel like a date picker would be beyond the scope of Stardust and would be better served as a plugin/extension implementation instead.

A date picker is essential to any component library.  That said, I'm not sure if it will live in this repo long term but we can certainly start it here.

For more context, in the long term Stardust intends to provide specifications and utilities for building compliant components.  In the short term, we're building lots of components to flesh out the specs and patterns.  At some point, we'll begin publishing these as separate packages.  Once that happens, we may have some low-level components here for ever.  It's not quite clear yet.

**Comment by brianespinosa** [comment]
@levithomason thanks for chiming in. You validated what I was kind of figuring would be the direction on this. It seems like a lot of the patterns and spec are still moving around a lot. This particular component seems like it would be one to wait on until those are more clear. I figured this RFC would be good more as an early placeholder.

> At some point, we'll begin publishing these as separate packages. Once that happens, we may have some low-level components here for ever. It's not quite clear yet.

Let me know when you get to the point of trying to make this move. I've got a lot of experience now working with Yarn workspaces at Espressive so I can probably help with the implementation (and point out some of the gotchas and pitfalls). The same workspaces structure should be possible with Lerna when we are ready to start publishing public repos of each component.
